rust/nix-expr/src/eval_state.rs

```diff
@@ -131,6 +137,46 @@ impl EvalState {
         }
         self.get_string(value)
     }
+    pub fn realise_string(
+        &self,
+        value: &Value,
+        is_import_from_derivation: bool,
+    ) -> Result<RealisedString> {
+        let t = self.value_type(value)?;
+        if t != ValueType::String {
+            bail!("expected a string, but got a {:?}", t);
+        }
+
+        let rs = unsafe {
+            raw::string_realise(
```

Comment:
It's not great, but ok for now.
To avoid this, the C API will have to expose the global logger so that we can set something else.

We should see this function as a high level helper anyway, because it is lacking some functionality that an interface for `Worker` (more like "scheduler") would offer, such as the ability to realise arbitrary sets of paths, or add more paths to realise while realisations are running.